Added virtual speaker approach support

PyrApple opened the pull request:
Following the discussion in the previous [pull-request](https://github.com/polarch/JSAmbisonics/pull/1). No modification to the Javascript library itself, simply added Matlab routines to create virtual speaker approach based Ambisonic HRIRs. Added example Matlab script showing how to compute direct / virtual HRIRs.

PyrApple committed:
Added Matlab routines to compute the ambisonic-to-binaural
decoding filters, using the virtual speaker approach

polarch commented:
Hi David,

I merged your contribution. I added additionally the HTML examples to the repository (easier to maintain and distribute), along with the original filters used in these first examples. More filters can be added and converted to examples for comparison. 

After I get back to Finland next week I'll commit a script that tries to find some reasonable left-right symmetric sets of directions from an HRTF set, with as few points as possible for a specified order (since it seems true that more points than needed add colouration). Then it can immediately pass the respective directions to your script and get the left-ear filters, then use these for the decoder.

I liked your demo with the reverb, it shows clearly to people that externalization does not depend much on HRTFs, no matter how perfect they are, it depends on reverberation (or decorrelation), as expected. A reverb object is on the TODO, but to be honest a true ambisonic reverb will require (N+1)^2 independent late RIR tails (they can be generated from a single RIR tail with decorrelation), which may be too much for the browser. But it's straightforward to do and good to have for completeness.

Alternatively, I'm planning to include a binaural reverb which is much cheaper (two RIRs plus a cross-talk network), since I assume most people will be using this with headphones.

Cheers,
Archontis

polarch commented:
Ah, I forgot to mention that I omitted the "sounds" folder from the examples, I have to check for a few of them where they're coming from before putting them publicly. But if you want a zip with all of them let me know and I can send you a download link on your e-mail or something.

PyrApple commented:
Dear Archontis,

Do not bother regarding the sounds, I already grabbed all of them from your dropbox folder.

The reverb I added is actually a "real ambisonic reverb", extracted from CATT with one of the LISTEN HRIR directly plugged in the room. To remain consistent with the decoder implementation, I assumed inter-aural axis symetry on the room as well as on the HRIR set. All in all, it's not that much intensive (but for the convolution size...) than anechoic binaural decoding. Still, +1 for the cheap binaural reverb (btw, [BinauralJS implementation](https://github.com/Ircam-RnD/binauralFIR)).

So, ready for the NodeJS pull-request? here it comes :).
